features/cryptocell/FEATURE_CRYPTOCELL310/sha512_alt.c

```diff
@@ -0,0 +1,86 @@
+/*
+ *  sha512_alt.c
+ *
+ *  Copyright (C) 2019, Arm Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "mbedtls/sha512.h"
+#if defined(MBEDTLS_SHA512_ALT)
+#include <string.h>
+#include "mbedtls/platform.h"
+
+void mbedtls_sha512_init( mbedtls_sha512_context *ctx )
+{
+    memset( ctx, 0, sizeof( mbedtls_sha512_context ) );
+
+}
+
+void mbedtls_sha512_free( mbedtls_sha512_context *ctx )
+{
+    if( ctx == NULL )
+        return;
+    CRYS_HASH_Free( &ctx->crys_hash_ctx );
+    memset( ctx, 0, sizeof( mbedtls_sha512_context ) );
+}
+
+void mbedtls_sha512_clone( mbedtls_sha512_context *dst,
+                           const mbedtls_sha512_context *src )
+{
+    memcpy(dst,src,sizeof(mbedtls_sha512_context));
+}
+
+
+int mbedtls_sha512_starts_ret( mbedtls_sha512_context *ctx, int is384 )
+{
+    if( is384 )
+        return( MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED );
+    if( CRYS_HASH_Init( &ctx->crys_hash_ctx, CRYS_HASH_SHA512_mode ) != CRYS_OK )
+        return ( MBEDTLS_ERR_PLATFORM_HW_ACCEL_FAILED );
+    return ( 0 );
+}
+
+int mbedtls_internal_sha512_process( mbedtls_sha512_context *ctx,
+                                     const unsigned char data[128] )
+{
+    return( MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED );
```

Comment:
Will this break applications that were using software SHA-512 previously? Should we return non-error here?

Reply:
no. 
This function is called only by our internal test.
Please look at https://github.com/ARMmbed/mbed-os/pull/8728 and discuss @yanesca for more information